executor/ddl.go

```diff
@@ -77,12 +112,42 @@ func (e *DDLExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	e.done = true
 
 	// For each DDL, we should commit the previous transaction and create a new transaction.
-	// An exception is create local temporary table.
-	if s, ok := e.stmt.(*ast.CreateTableStmt); ok {
```

Comment:
May be you can make a function `func dropLocalTemporaryTables([]*ast.TableName)` and move local temp tables out of the DropTableStmt and check the left tables' count. If the count is zero, call `dropLocalTemporaryTables` immediately and then return. If statement still have normal tables, call dropLocalTemporaryTables` after dropping normal tables. It seems more clearly.